source/material_model/rheology/strain_dependent.cc

```diff
@@ -556,75 +556,113 @@ namespace aspect
 
             // We need to obtain the strain values from compositional fields at the previous time step,
             // as the values from the current linearization point are an extrapolation of the solution
-            // from the old timesteps. First, create arrays to store these values
-            std::vector<double> old_plastic_strain(in.n_evaluation_points());
-            std::vector<double> old_viscous_strain(in.n_evaluation_points());
-            std::vector<double> old_total_strain(in.n_evaluation_points());
-            std::vector<double> old_noninitial_plastic_strain(in.n_evaluation_points());
-
+            // from the old timesteps.
             // Prepare the field function and extract the old solution values at the current cell.
-            Functions::FEFieldFunction<dim, LinearAlgebra::BlockVector>
-            fe_value(this->get_dof_handler(), this->get_old_solution(), this->get_mapping());
-            fe_value.set_active_cell(in.current_cell);
+            std::vector<Point<dim>> quadrature_positions(1,this->get_mapping().transform_real_to_unit_cell(in.current_cell, in.position[i]));
+            boost::container::small_vector<double, 100> old_solution_values(this->get_fe().dofs_per_cell);
+            in.current_cell->get_dof_values(this->get_old_solution(),
+                                            old_solution_values.begin(),
+                                            old_solution_values.end());
+
+            if (composition_evaluators.size() == 0)
+              composition_evaluators.resize(this->n_compositional_fields());
+
+            Assert(composition_evaluators.size() == this->n_compositional_fields(),
```

Comment:
I'm a bit confused by this Assert. composition_evaluators is created with 
`mutable std::vector<std::unique_ptr<FEPointEvaluation<1, dim>>> composition_evaluators;`. 

In this instance, would the above if statement be triggered? - ( `if (composition_evaluators.size() == 0)`). If not, when composition_evaluators be of size 0 and the resize line (`composition_evaluators.resize(this->n_compositional_fields());`) be triggered?

In summary, I'm struggling a bit to understand when and why this variable should be adjusted. A bit more documentation may help and apologies if I am just overlooking something obvious.